guides/controllers/params-validation.md

```diff
@@ -1,52 +1,149 @@
+
+### Benefits
+
+- Expression and explicitnes about the parameters the model exptects.
+- Security by whitelisting only the parameters allowed per action.
+- Data correctnes to prevent invalid inputs to propagate in the system.
+
+### Example Usage
 
 ```ruby
 class UsersController < ApplicationController
   def create
-    unless params.valid?
-      response.puts {errors: params.errors}.to_json
-      response.status_code 400
+    user = User.new(user_params.validate!)
+    
+    if user.save
+      redirect_to action: :index, flash: {:success => "Created user successfully!"}
+    else
+      redirect_to action: :index, flash: {:error => "Could not create user!"}
     end
+  end
+  
+  def user_params
+    params.validation do
+     required(:name, "Your First Name is missing!") { |p| p.name? & !p.name.empty? }
+     required(:email, "Your email address is invalid!") { |p| p.email? & p.size.between? 1..10 }
+     required(:last_name) { |p| p.last_name? }
+    end
+  end
+end
+```
 
-    user = User.new params.validate!
-    user.save!
+### Validation API
 
-    @client = Client.new
-    redirect_to :index
-  end
+`#validation` Setups validation rules to be performed.
 
-  # Define parameters within actions
-  def update
-    update_params = params.validation do
-      required(:name, "Your First Name is missing!") { |p| p.name? & !p.name.empty? }
-      required(:email, "Your email address is invalid!") { |p| p.email? & p.size.between? 1..10 }
-      required(:last_name) { |p| p.last_name? }
-    end
+- Use `#required(field)` method to define require fields.
+- Use `#optional(field)` method to define optional fields.
 
-    unless update_params.valid?
-      response.puts {errors: params.errors}.to_json
-      response.status_code 400
-    end
+```ruby
+params.validation do
+  required(:email) { |p| p.url? }
+  optional(:age)
+end
+```
 
-    user = User.new update_params.validate!
-    user.save!
+`#validate!` Input must be valid otherwise raises error, if valid returns a hash of validated params Otherwise raises a Validator::ValidationFailed error messages contain errors.
 
-    redirect_to :index
-  end
+```ruby
+user = User.new params.validate!
+```
+
+`#valid?` Returns True or false whether the validation passed
+
+```ruby
+unless params.valid?
+  response.puts {errors: params.errors}.to_json
+  response.status_code 400
 end
 ```
 
+`#errors` Returns an array of errors for the invalid inputs. This array of errors is populated after running `#validate!` or `#valid?`.
+
+```ruby
+params.errors
+```
+
+### Field Validation Rules
+
+Amber has extrended the Crystal String and Number classes with additional methods to assit with better validation.
+
+| String                      | Number          |
+|-----------------------------|-----------------|
+| str?                        | positive?       |
```

Comment:
@eliasjpr Can you surround all these methods with backtrics? like this `positive?` :sweat_smile: